rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/RMNodeStarter.java

```diff
@@ -502,7 +495,14 @@ private boolean allNodesAreAvailable(List<Node> nodes, ResourceManager rm) {
             if (!rm.setNodeAvailable(url).getBooleanValue()) {
                 if (!rm.nodeIsAvailable(url).getBooleanValue()) {
                     logger.info("Node " + url + " removed");
-                    iterator.remove(); // node removed manually by user
+                    // node removed manually by user
+                    // kill the local worker node
+                    try {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)